tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2214,69 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def matrix_profile(x, param):
+    """
+    TODO: Documentation
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries {"sample_pct": x, "threshold": y, "feature": z}
+    with sample_pct and threshold being parameters of the matrixprofile
+    package https://matrixprofile.docs.matrixprofile.org/api.html#matrixprofile-compute
+    and feature being one of "min", "max", "mean", "median", "25", "75"
+    and decides which feature of the matrix profile to extract
+    :type param: list
+    :return: the different feature values
+    :return type: pandas.Series
+    """
+    x = np.asarray(x)
+
+    def _calculate_mp(**kwargs):
+                """Calculate the matrix profile using the specified window, or the maximum subsequence if no window is specified"""
+        try:
+            if 'windows' in kwargs:
+                m_p = mp.compute(x,**kwargs)
+
+            else:
+                m_p = mp.algorithms.maximum_subsequence(x, include_pmp=True)['pmp'][-1]
+            return m_p[(~np.isnan(m_p)) & (~np.isinf(m_p))]
```

Comment:
I don't think you want to return a modified version of the matrix profile at this stage. What if additional features in the future handle imputation or something?

@nils-braun based on our conversation in the matrixprofile github issue, how do you want to handle any exception vs the specific "NoSolutionPossible" case?
```
    return m_p
except mp.exceptions.NoSolutionPossible as e:
    warnings.warn(str(e))
    return None
except Exception as e:
    # ?????????
    return None
```

Reply:
Yes, I think a warning would make sense here! The question is, do we expect any other exception? If not, let's do not catch it and let it actually fail for the user